industrydive/sourcedive
# TECH-8227 prevent user from editing interaction created by someone else

[TECH-8227 prevent user from editing interaction created by someone else](https://industrydive.atlassian.net/browse/TECH-8227)

### Non-technical Context
Prevent user from editing interaction created by someone else unless the creator has listed that other user as an interviewer.

### Technical Context
Update the existing read-only logic for `admin.py` to reflect this.

### Reproduction Steps
- checkout develop
- visit http://127.0.0.1:8080/admin/sources/interaction/
- make sure there are at least two interactions that you can see that were created by another user (not you): one public and one semi-private
- if not, log in incognito as another user and create those two
- select the public interaction not created by you
- click "edit" in the top right
- [x] verify you are able to edit it
- repeat the same for the steps for a semi-private interaction
- [x] verify you are able to edit it

### Manual Testing Instructions
- checkout this branch
- visit http://127.0.0.1:8080/admin/sources/interaction/
- select the public interaction created by another user
- click "edit" in the top right
- [x] verify that you are not able to edit it
- select the semi-private interaction created by another user
- click "edit" in the top right
- [x] verify that you are not able to edit it
- edit the semi-private interaction as the other user
- add yourself as an interviewer on that interaction
- click "save and continue editing"
- switch back to your user
- view that same interaction
- [x] confirm you can edit the interaction
- [x] confirm you can view the notes field

<img width="904" alt="Screen Shot 2020-05-15 at 4 07 29 PM" src="https://user-images.githubusercontent.com/327783/82091894-37ad0b80-96c6-11ea-9058-fa4ee2dea89e.png">

- switch back to the other user
- remove yourself as an interviewer
- click "save and continue editing"
- switch back to your user
- refresh or reopen that semi-private interaction
- [x] confirm you can not longer edit it
- [x] confirm you cannot see the note

<img width="969" alt="Screen Shot 2020-05-15 at 4 08 05 PM" src="https://user-images.githubusercontent.com/327783/82091928-48f61800-96c6-11ea-9756-796e2e9be4a1.png">

### QA Checklist

#### Developer

    - [n/a] Technical documentation


#### Reviewer

    - If there are few or minor issues, add inline GitHub comments
    - If there are many issues or it seems like the developer did not follow the style guide, leave a comment asking them to do so
